src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/check_health.py

```diff
@@ -0,0 +1,266 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+from .custom import get_docker_command
+from ._docker_utils import _get_aad_token
+from .helm import get_helm_command
+from ._utils import get_registry_by_name
+from ._errors import ErrorClass
+
+
+DOCKER_PULL_SUCCEEDED = "Downloaded newer image for {}"
+DOCKER_IMAGE_UP_TO_DATE = "Image is up to date for {}"
+IMAGE = "mcr.microsoft.com/mcr/hello-world:latest"
+FAQ_MESSAGE = "\nPlease refer to https://aka.ms/acr/faq for more informations."
+
+
+# Utilities functions
+
+
+def print_pass(message):
+    from colorama import Fore, Style, init
+    init()
+    print(str(message) + " : " + Fore.GREEN + "OK" + Style.RESET_ALL)
+
+
+def print_warning(message):
+    from colorama import Fore, Style, init
+    init()
+    print(Fore.YELLOW + str(message) + Style.RESET_ALL)
+
+
+def print_error(message):
+    from colorama import Fore, Style, init
+    init()
+    print(Fore.RED + str(message) + Style.RESET_ALL)
+
+
+def _handle_error(error, ignore_errors):
+    if ignore_errors:
+        print_error(error.get_error_message())
+    else:
+        raise CLIError(error.get_error_message(FAQ_MESSAGE))
+
+
+def _subprocess_communicate(command_parts, shell=False):
+    from subprocess import PIPE, Popen, CalledProcessError
+    output, stderr = "", ""
+
+    try:
+        p = Popen(command_parts, stdout=PIPE, stderr=PIPE, shell=shell)
+        output, stderr = p.communicate()
+        output = output.decode('UTF-8').rstrip()
+        stderr = stderr.decode('UTF-8').rstrip()
+    except CalledProcessError as e:
+        stderr = str(e)
+
+    if stderr:
+        stderr = "Failed to run command '{}'. {}".format(
+            ' '.join(command_parts),
+            stderr
+        )
+    return output, stderr
+
+
+# Checks for the environment
+
+
+# Checks docker command, docker daemon, docker version and docker pull
+def _get_docker_status_and_version(ignore_errors, yes):
+    from ._errors import DOCKER_DAEMON_ERROR, DOCKER_PULL_ERROR, DOCKER_VERSION_ERROR
+
+    # Docker command and docker daemon check
+    docker_command, error = get_docker_command(is_diagnostics_context=True)
+    docker_daemon_available = True
+
+    if error:
+        _handle_error(error, ignore_errors)
+        if error.error_title != DOCKER_DAEMON_ERROR.error_title:
+            return  # We cannot proceed if the error is unexpected or with docker command
+        else:
+            docker_daemon_available = False
+
+    if docker_daemon_available:
+        print("Docker daemon status: available")
+
+    # Docker version check
+    output, stderr = _subprocess_communicate([docker_command, "--version"])
+    if stderr:
```

Comment:
Same here. Some warnings regarding config file loading can potentially be ignored. Such as: `WARNING: Error loading config file: /xyz/.docker/config.json: stat /xyz/.docker/config.json: permission denied`